bfvmm/src/intrinsics/tests/test_apic_intel_x64_x2apic.cpp

```diff
@@ -1508,6 +1508,58 @@ TEST_CASE("x2apic_supported")
     CHECK_FALSE(x2apic::supported());
 }
 
+TEST_CASE("x2apic_control_validate_gpa_op")
```

Comment:
For the validate_{gpa,msr}_op tests, we need to test both x2apic and xapic for every "troublesome" register. So add tests for each register that "diverges" between xapic and x2apic. So for example, you would have tests for reading and writing each ICR register for both x2apic and xapic. x2apic would fail for any gpa op to the low ICR register. Another is the APR - all operations on the x2apic would fail because this register wouldn't exist.